napari/animation/animate.py

```diff
@@ -0,0 +1,303 @@
+        else:
+            inter_steps = self.inter_steps
+
+        empty = {
+            "ndisplay": [],
+            "frame": [],
+            "camera": [],
+            "vis": [],
+            "sliders": [],
+        }
+        states_dict = []
+        for ind, x in enumerate(self.key_frames):
+            states_dict.append(x)
+            # do not add frames after last key-frame
+            if ind < len(self.key_frames) - 1:
+                # do not add frames when switching camera
+                if (
+                    self.key_frames[ind]['ndisplay']
+                    == self.key_frames[ind + 1]['ndisplay']
+                ):
+                    for y in range(inter_steps[ind]):
+                        states_dict.append(copy.deepcopy(empty))
+        for ind, x in enumerate(states_dict):
+            x["frame"] = ind
+        self.states_dict = states_dict
+
+    def create_steps(self):
+        """Interpolate states between key-frames"""
+
+        self.create_state_dict()
+        self.interpolated_states = util.interpolate(self.states_dict)
+
+    def key_interpolframe(self, viewer):
+        """Progress through interpolated frames"""
+
+        self.create_steps()
+
+        new_frame = (self.current_interpolframe + 1) % len(self.states_dict)
+        self.update_napari_state(new_frame)
+        self.current_interpolframe = new_frame
+
+    def collect_images(self):
+        """Collect images corresponding to all interpolated states
+
+        Returns
+        -------
+        image_stack : 3D numpy
+            stack of all snapshots
+        """
+
+        images = []
+        self.create_steps()
+        for i in range(len(self.states_dict)):
+
+            self.update_napari_state(i)
+            images.append(self.viewer.screenshot())
+
+        image_stack = np.stack(images, axis=0)
+        return image_stack
+
+    def update_napari_state(self, frame):
+        """Set the napari viewer to a given interpolated frame
+
+        Parameters
+        -------
+        frame : int
+            frame to visualize
+        """
+
+        # set view type 2D/3D and camera state
+        self.viewer.dims.ndisplay = self.interpolated_states["ndisplay"][frame]
+        self.viewer.window.qt_viewer.view.camera.set_state(
+            self.interpolated_states["camera"][frame]
+        )
+
+        # assign interpolated visibility state
+        for j in range(len(self.viewer.layers)):
+            self.viewer.layers[j].visible = self.interpolated_states["vis"][
+                frame
+            ][j]
+
+        # adjust slider positions
+        for i in range(self.interpolated_states["sliders"].shape[1]):
+            self.viewer.dims.set_point(
+                i, self.interpolated_states["sliders"][frame][i]
+            )
+
+        # update view
+        self.viewer.window.qt_viewer.view.camera.view_changed()
+
+    def make_movie(self, name="movie.mp4", fps=20, quality=5):
```

Comment:
can you also expose a keyword arg `format=None` that get's passed on to `get_writer` - see [function signature here](https://imageio.readthedocs.io/en/stable/userapi.html#imageio.get_writer).